src/components/GalleryBrick/index.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+import FontAwesome from 'react-fontawesome';
+import styles from './styles';
+
+class GalleryBrick extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      pages: this.props.pages || 1,
+      currentPage: this.props.currentPage || 1,
+      pageImages : Math.round(this.props.images.length / this.props.pages)
+    }
+  }
+
+  renderImages(src, s, key){
+    return (
+      <div style={ s.imgGallery }>
+        <img key={ key } style={ s.image} src={ src } />
+      </div>
+      )
+  }
+
+  renderPagesImages(images, s){
+    if (!images) return null;
+    let boxImages = [], max = (this.state.currentPage * this.state.pageImages) - this.state.pageImages;
+
+    for( let i = max ; i <= this.state.pageImages + max - 1 ; i++ ){ 
+      boxImages.push(images[i]);
+    };
+    return boxImages.map((img, key) => this.renderImages(img, s, key));
+  }
+
+  renderPrevBtn(s){
+    if(this.state.currentPage === 1) return null;
+    return (
+      <a 
+        style={ s.controls } 
+        onClick={ () => this.setState({ currentPage : this.state.currentPage - 1 }) } >
+        <FontAwesome name="chevron-left" />
+      </a>
+    )
+  }
+
+  renderNextBtn(s){
+    if(this.state.currentPage >= this.state.pages) return null;
+    return (
+      <a 
+        style={ s.controls } 
+        onClick={ () => this.setState({ currentPage : this.state.currentPage + 1 }) } >
+        <FontAwesome name="chevron-right" />
+      </a>
+    )
+  }
+
+  renderIndicators(s){
+    let { pages, currentPage } = this.state, count = [...Array(pages).keys()];
+
+    if(pages < 2) return null;
+
+    return count.map((item) =>{
+      return (
+        <FontAwesome 
+          style={ s.indicator } 
+          onClick={ ()=> this.setState({ currentPage : item + 1 }) } 
+          name={ (currentPage === item + 1) ? 'circle' : 'circle-o' } 
+        />
+      )
+    });
+  }
+
+  render() {
+
+    let { images, title, description, type } = this.props, s = styles(this.props);
```

Comment:
- what is the "type" prop for?
- images prop should be names items
- itemsPerPage should be set through props to define the quantity of items in each page